### PR TITLE
Implement get program endpoint

### PR DESCRIPTION
**Pull request by brandonwong368**
### Notion ticket link

[Ticket Name](https://www.notion.so/uwblueprintexecs/086c1b564dbf478faa649a8880e34b81?v=9d3f8002a0d1477ba395616dfc082af7&p=eb4f61da390f40b98d2321e90a5b3126)

### Implementation description

-add `getProgram` and `getPrograms` functions as part of the GET program endpoints

### Steps to test

1.

### What should reviewers focus on?

-

### Checklist

-   [ ] My PR name is descriptive and in imperative tense
-   [ ] I have run the linter
-   [ ] I have requested a review from the PL, and/or other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: pages/api/program/index.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ */
+export default async function handle(
+    req: NextApiRequest,
+    res: NextApiResponse,
+): Promise<void> {
+    switch (req.method) {
+        case "GET": {
+            const programs = await getPrograms();
+            ResponseUtil.returnOK(res, programs);
+            res.status(200).json({});
```
Comment (feng-eric) [inline pages/api/program/index.ts]:
```suggestion
    
```

Comment (feng-eric) [reply pages/api/program/index.ts]: Remove this line as it is being done in `returnOK`

File: pages/api/program/index.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            // TODO:
+            break;
+        }
+        case "DELETE": {
+            // TODO:
+            break;
+        }
+        default: {
+            const allowedHeaders: string[] = ["GET", "POST", "PUT", "DELETE"];
+            // TODO: add JSON response for method not allowed
```
Comment (feng-eric) [inline pages/api/program/index.ts]: Remove TODO

File: services/database/program.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ * getProgram takes the id parameter and returns the program associated with the programId
+ * @param {string} id - programId
+ */
+
+async function getProgram(id: string) {
+    const program = await prisma.program.findUnique({
+        where: {
+            id: parseInt(id),
+        },
+        //get program() exposes CRUD operations for Program model
```
Comment (feng-eric) [inline services/database/program.ts]: Remove comments

File: services/database/program.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        //potentially include programs : true ?
+        //include: {},
+    });
+    return program;
+}
+
+async function getPrograms() {
+    const programs = await prisma.program.findMany({
+        //include: {},
```
Comment (feng-eric) [inline services/database/program.ts]: Remove comment

**Review by feng-eric — CHANGES_REQUESTED** [review]
Had a few comments!

File: services/database/program.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import prisma from "@database";
+
+/**
+ * NOTE: https://www.prisma.io/docs/concepts/components/prisma-client/advanced-type-safety/operating-against-partial-structures-of-model-types
+ * getProgram takes the id parameter and returns the program associated with the programId
+ * @param {string} id - programId
+ */
+
+async function getProgram(id: string) {
```
Comment (jason17h) [inline services/database/program.ts]: I think return type should be `Promise<Program>`

File: pages/api/program/[id].ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { ResponseUtil } from "@utils/responseUtil";
+import { getProgram } from "@database/program";
+// TODO: Type the response data
```
Comment (feng-eric) [inline pages/api/program/[id].ts]: This TODO still required?

File: pages/api/program/[id].ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+ */
+export default async function handle(
+    req: NextApiRequest,
+    res: NextApiResponse,
+): Promise<void> {
+    if (req.method == "GET") {
+        // Obtain user id
+        const { id } = req.query;
+
+        // obtain user with provided userId
```
Comment (feng-eric) [inline pages/api/program/[id].ts]:
```suggestion
        // obtain program with provided programId
```

File: services/database/program.ts [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+        where: {
+            id: parseInt(id),
+        },
+        //get program() exposes CRUD operations for Program model
+        //potentially include programs : true ?
+        //include: {},
+    });
+    return program;
+}
+
```
Comment (feng-eric) [inline services/database/program.ts:17]: Add code documentation

**Comment by feng-eric** [comment]
Hey @brandonwong368, I think you requested review but you didn't update the code.

File: services/database/program.ts [hunk 9]
```diff
@@ -0,0 +1,27 @@
+import prisma from "@database";
+import { Program } from "@prisma/client";
+/**
+ * NOTE: https://www.prisma.io/docs/concepts/components/prisma-client/advanced-type-safety/operating-against-partial-structures-of-model-types
+ * getProgram takes the id parameter and returns the program associated with the programId
+ * @param {string} id - programId
+ *
+ */
+
```
Comment (feng-eric) [inline services/database/program.ts]:
```suggestion
```

**Review by feng-eric — APPROVED** [review]
LGTM! Good work :)

**Pull request merged**